discovery-infra/test_infra/utils/env_variables_utils.py

```diff
@@ -0,0 +1,87 @@
+from distutils.util import strtobool
+from pathlib import Path
+from typing import List
+
+from dataclasses import dataclass, field
+
+from test_infra import consts
+from test_infra.consts import resources, env_defaults
+from test_infra.utils import operators_utils, get_env, get_openshift_version
+from test_infra.utils.cluster_name import ClusterName
+
+
+def get_kubeconfig_path(cluster_name: ClusterName):
+    kubeconfig_dir = env_defaults.DEFAULT_KUBECONFIG_DIR
+    default = kubeconfig_dir.joinpath(f"kubeconfig_{cluster_name}")
+    kubeconfig_path = get_env("KUBECONFIG", str(default))
+
+    if kubeconfig_path == str(default):
+        kubeconfig_dir.mkdir(parents=True, exist_ok=True)
+
+    return kubeconfig_path
+
+
+@dataclass(frozen=True)
+class EnvVariablesUtils:
+    ssh_public_key: str = get_env("SSH_PUB_KEY")
+    remote_service_url: str = get_env("REMOTE_SERVICE_URL")
+    pull_secret: str = get_env("PULL_SECRET")
+    offline_token: str = get_env("OFFLINE_TOKEN")
+    openshift_version: str = get_openshift_version()
+    base_dns_domain: str = get_env("BASE_DOMAIN", consts.DEFAULT_BASE_DNS_DOMAIN)
+    masters_count: int = int(get_env("MASTERS_COUNT", get_env("NUM_MASTERS", env_defaults.DEFAULT_NUMBER_OF_MASTERS)))
+    workers_count: int = int(get_env("WORKERS_COUNT", get_env("NUM_WORKERS", env_defaults.DEFAULT_WORKERS_COUNT)))
+    nodes_count: int = masters_count + workers_count
+    num_day2_workers: int = int(get_env("NUM_DAY2_WORKERS", env_defaults.DEFAULT_DAY2_WORKERS_COUNT))
+    vip_dhcp_allocation: bool = bool(strtobool(get_env("VIP_DHCP_ALLOCATION")))
+    worker_memory: int = int(get_env("WORKER_MEMORY", resources.DEFAULT_WORKER_MEMORY))
+    master_memory: int = int(get_env("MASTER_MEMORY", resources.DEFAULT_MASTER_MEMORY))
+    network_mtu: int = int(get_env("NETWORK_MTU", resources.DEFAULT_MTU))
+    worker_disk: int = int(get_env("WORKER_DISK", resources.DEFAULT_WORKER_DISK))
+    master_disk: int = int(get_env("MASTER_DISK", resources.DEFAULT_MASTER_DISK))
+    master_disk_count: int = int(get_env("MASTER_DISK_COUNT", resources.DEFAULT_DISK_COUNT))
+    worker_disk_count: int = int(get_env("WORKER_DISK_COUNT", resources.DEFAULT_DISK_COUNT))
+    storage_pool_path: str = get_env("STORAGE_POOL_PATH", env_defaults.DEFAULT_STORAGE_POOL_PATH)
+    cluster_name: ClusterName = ClusterName()
```

Comment:
class?

Reply:
I know, But we need to handle both suffix and prefix on some cases (without parsing it every time).
I discussed about it with @YuviGold few days ago